dll/win32/iernonce/registry.cpp

```diff
@@ -0,0 +1,211 @@
+/*
+ * PROJECT:     ReactOS system libraries
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Functions to read RunOnceEx registry.
+ * COPYRIGHT:   Copyright 2021 He Yang <1160386205@qq.com>
+ */
+
+
+#include <cassert>
+#include "registry.h"
+#include <winreg.h>
+
+
+typedef BOOL(*REGENUMCALLBACK)(CRegKeyEx &hKey, LPCWSTR lpObjectName, DWORD cchszName, PVOID Param);
+
+enum class REG_ENUM_TYPE
+{
+    SUBKEY,
+    VALUE
+};
+
+
+LONG CRegKeyEx::EnumValueName(
+    DWORD iIndex,
+    LPTSTR pszName,
+    LPDWORD pnNameLength)
+{
+    return RegEnumValueW(m_hKey, iIndex, pszName, pnNameLength,
+        NULL, NULL, NULL, NULL);
+}
+
+bool RunOnceExSectionCmp::operator()(const pRunOnceExSection &lhs, const pRunOnceExSection &rhs) const
+{
+    return lstrcmpW(lhs->m_SectionName.GetString(), rhs->m_SectionName.GetString()) < 0;
+}
+
+bool RunOnceExEntryCmp::operator()(const ATL::CStringW &lhs, const ATL::CStringW &rhs) const
```

Comment:
Where is this going to be used?

Reply:
this and the code above is used to sort these stuffs in the container (`std::set` and `std::map`), I guess we would want these stuff being displayed & executed by string order

Reply:
Ok, then this needs some comments in the code.